build_tools/github_actions/start_memory_monitor.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+# Script to start memory monitoring for build process
+# This script starts the memory monitor in the background and captures its PID
+
+# Set default values if environment variables are not set
+BUILD_DIR="${BUILD_DIR:-build}"
+JOB_NAME="${JOB_NAME:-default}"
+PHASE="${PHASE:-Build Phase}"
+
+# Create logs directory in BUILD_DIR
+mkdir -p "$BUILD_DIR/logs"
+
+# Start memory monitor in background
+python build_tools/memory_monitor.py \
+  --phase "$PHASE" \
+  --log-file "$BUILD_DIR/logs/build_memory_log_${JOB_NAME}.jsonl" \
+  --background > "$BUILD_DIR/logs/monitor_output_${JOB_NAME}.txt" 2>&1 &
```

Comment:
These could discover the repository root themselves instead of assuming they will be run with CWD matching the repository root. See https://github.com/ROCm/TheRock/blob/main/docs/development/style_guide.md#dont-make-assumptions-about-the-current-working-directory